packages/core/src/browser/frontend-idle-status.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, optional } from 'inversify';
+import { ILogger } from '../common/logger';
+import { Event, Emitter } from '../common/event';
+import { FrontendApplicationContribution } from './frontend-application';
+
+/**
+ * Service providing information whether the application is idle or not.
+ */
+export const FrontendIdleService = Symbol('ApplicationIdleService');
+export interface FrontendIdleService {
+
+    /**
+     * The current idle state of the application. `true` is idle, `false` is active.
+     */
+    readonly idle: boolean;
+
+    /**
+     * Clients could listen on the idle state of the frontend application.
+     */
+    readonly onStatusChange: Event<boolean>;
+
+}
+
+@injectable()
+export class IdleStatusOptions {
+
+    /**
+     * The default idle options. The application goes into idle state after 15 seconds.
+     */
+    static DEFAULT: IdleStatusOptions = {
+        idleAfter: 15000
```

Comment:
I like the idea of stopping the connection status checks when the frontend is idle (e.g. cloud instance that might be used sporadically). However I find 15 sec to be a short delay to be considered idle. If it were configurable, I would not mind. I would suggest 1 minute or more instead.

In my tests, I can lose the connection, and the idle mode kicks-in before it's detected that the connection was lost (assuming I do not move the mouse in the browser window or type in it).

Reply:
I also noticed that with the current idle timer, idle mode can kick-in while the user is patiently waiting for the connection to be detected as being up again. In that case, unless I generate bogus activity to prevent Theia from going into idle mode, it will never be detected that the connection has recovered.

e.g. in this video the connection is available when I dismiss the notification: 

![peek 2018-02-06 07-11](https://user-images.githubusercontent.com/25749063/35859139-f0a34688-0b0c-11e8-9015-b800925ab84a.gif)

Reply:
Thanks! I will split up the PR, open a follow-up and leave the idle feature out.

Reply:
Ok, if you think there need to be more discussions about the length of the idle timer. Else, with a longer timer, I think it would work fine.

Reply:
I think it is a useful and cool feature we have to support; I just need to wrap up other tasks too.

Reply:
@kittaakos please keep it in and just increase the idle timeout to the greater than the connection timeout , @marcdumais-work also agrees with this

Reply:
Sure, if you guys like this feature that much, we keep it as part of this PR :)

Reply:
Thanks @kittaakos - it is a nice feature.

Reply:
The credit goes to @hexa00.

Reply:
I have (force) pushed the changes regarding the idle service. Could you please verify the UX again? Thank you!

Reply:
ok, I will try it right away

Reply:
Works well, thanks @kittaakos